src/test/coins_tests.cpp

```diff
@@ -154,9 +154,16 @@ void SimulationTest(CCoinsView* base, bool fake_best_block)
             bool test_havecoin_after = InsecureRandBits(2) == 0;
 
             bool result_havecoin = test_havecoin_before ? stack.back()->HaveCoin(COutPoint(txid, 0)) : false;
-            const Coin& entry = (InsecureRandRange(500) == 0) ? AccessByTxid(*stack.back(), txid) : stack.back()->AccessCoin(COutPoint(txid, 0));
+
+            // Infrequently, test usage of AccessByTxid instead of AccessCoin - the
+            // former just delegates to the latter and returns the first unspent in a txn.
+            const Coin& entry = (InsecureRandRange(500) == 0) ?
```

Comment:
In the second commit 6d8affca96c7a34f5f104 while adding the explanatory comment, it may be worth describing why it is run infrequently.  If I run `./src/test/test_bitcoin -t coins_tests` with the following change, the test run appears to hang (or take a very long time) with this loop being run `NUM_SIMULATION_ITERATIONS = 40000` iterations.  

```diff
            // Infrequently, test usage of AccessByTxid instead of AccessCoin - the
            // former just delegates to the latter and returns the first unspent in a txn.
-            const Coin& entry = (InsecureRandRange(500) == 0) ?
-                AccessByTxid(*stack.back(), txid) : stack.back()->AccessCoin(COutPoint(txid, 0));
+            const Coin& entry = AccessByTxid(*stack.back(), txid);
```

6d8affca96c7a34f5f104 nit, can drop redundant brackets while touching the line

```suggestion
            const Coin& entry = InsecureRandRange(500) == 0 ?
```